catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -29,8 +33,20 @@ def update_state(_):
             break
     for sensor in sensor_priorities["z"]:
         if sensor.isActive():
-            z = sensor.z
-            break
+            if sensor.sensor_name == "depth_sensor":
+                ds_z = sensor.z
+                if dvl_z is not None:
+                    offset = math.abs(ds_z - dvl_z)
+                    i += 1
+                    ds_z -= math.exp(-i)*offset
+                z = ds_z
+            else:
+                dvl_z = sensor.z
+                if ds_z is not None:
+                    offset = math.abs(ds_z - dvl_z)
+                    dvl_z -= offset
+                z = dvl_z
```

Comment:
Could you also include comments for what is going on here and why? I think it would make it easier for you to debug and also for me to review